lib/component-repository/src/routes/virtual-components/CreateComponentConfig.js

```diff
@@ -1,3 +1,36 @@
+const ComponentConfig = require('../../models/ComponentConfig');
+const mongoose = require('mongoose');
+
 module.exports = async function (req, res, next) {
-    return next();
+  const { authClientId } = req.body;
+  const { componentVersionId } = req.params;
+  const tenant = req.user.isAdmin ? req.body.tenant : req.user.tenant;
+
+  if (!mongoose.Types.ObjectId.isValid(componentVersionId)) {
+    return res
+      .status(400)
+      .send({ errors: [{ message: 'Invalid id', code: 400 }] });
+  }
+  const currentVersionCompConfig = await ComponentConfig.findOne({
+    componentVersionId,
+    tenant,
+  });
+
+  if (currentVersionCompConfig) {
+    return res.status(400).send({
+      errors: [
+        { message: 'This component version has already been set.', code: 400 },
+      ],
+    });
+  }
+
+  const newConfig = new ComponentConfig({
+    authClientId,
+    componentVersionId,
+    tenant
+  });
+
```

Comment:
After you create the mongoose object, it would be better if you call the validate( function like in other places to bbe sure that there are no errors

Reply:
Done